backend/src/modules/users/users.service.ts

```diff
@@ -0,0 +1,33 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { CreateUserDto } from './dto/create-users.dto';
+import * as argon2 from 'argon2';
+import { UserRepository } from './users.repository';
+
+@Injectable()
+export class UsersService {
+  constructor(private readonly userRepository: UserRepository) {}
+
+  async create(createUserDto: CreateUserDto) {
+    const existUser = await this.userRepository.findOne({
+      where: {
+        email: createUserDto.email,
+      },
+    });
+    if (existUser) throw new BadRequestException('This email already exist');
+
+    const user = await this.userRepository.createUser({
+      firstName: createUserDto.firstName,
+      lastName: createUserDto.lastName,
+      email: createUserDto.email,
+      age: createUserDto.age,
+      password: await argon2.hash(createUserDto.password),
+      balance: createUserDto.balance,
+      roleId: createUserDto.roleId,
```

Comment:
If we don't have this property(because it's optional) on the createUserDto our server will crush